Source/External/ExCSS/Parser.Blocks.cs

```diff
@@ -554,7 +554,7 @@ private bool ParseSingleValueHexColor(string color)
 
             if(HtmlColor.TryFromHex(color, out htmlColor))
                 return AddTerm(htmlColor);
-            return false;
+			throw new ArgumentOutOfRangeException(nameof(color), $"{color} is not a valid hex color");
         }
```

Comment:
if we raise here, we have to handle the excption in the caller code, and use the default color instead. Or we may do the conversion here, if we have enough information (not sure, I haven't checked the code), and issue a warning instead of the exception.

Reply:
Hm, I thought I had already explained why I don't want to have the exception here, but can't find it now. Just in case it got lost, or I actually didn't click the Comment button: 
We usually try to behave like browsers, and browsers in this case (e.g. incorrect color description) just ignore the color and use the default. I recently fixed a bug along these lines. So we just haveto make sure that the default color is used, if the provided color is invalid.

Reply:
I reverted the change and changed my unit test, however if I feed an incorrect hex value the lexer goes haywire somewhere else (test: Lexer_FileWithInvalidHex_ColorTagIsIgnored)

```System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index```

I think that's why I introduced the "nicer" exception, since the lexer did crash anyway :D, so why not fail with a nice and descriptive exception. For now I flagged the test inconclusive to pass the build.